metomi/isodatetime/data.py

```diff
@@ -531,60 +646,76 @@ def __floordiv__(self, other):
                 "'%s' should be integer." %
                 type(other).__name__
             )
-        new = self.copy()
+        new = self._copy()
         if self.get_is_in_weeks():
-            new.weeks //= other
+            new._weeks //= other
             return new
-        new.years //= other
-        new.months //= other
-        new.days //= other
-        new.hours //= other
-        new.minutes //= other
-        new.seconds //= other
+        new._years //= other
+        new._months //= other
+        new._days //= other
+        new._hours //= other
+        new._minutes //= other
+        new._seconds //= other
         return new
 
-    def __eq__(self, other: "Duration") -> bool:
-        my_data = self.get_days_and_seconds()
-        other_data = other.get_days_and_seconds()
-        return my_data == other_data
+    def __hash__(self) -> int:
+        # TODO: alt calendar modes
+        if self.get_is_in_weeks():
+            return hash((0, 0, self._get_non_nominal_seconds()))
+        return hash(
+            (self._years, self._months, self._get_non_nominal_seconds()))
```

Comment:
Could this return the same value for a duration in weeks as one which isn't?

Since we are already performing the check, perhaps:

```python
return hash((self.get_is_in_weeks(), self._years, self._months, self._get_non_nominal_seconds()))
```

Reply:
`hash(P1W) == hash(P7D)`, seeing as `P1W == P7D`. Including `get_is_in_weeks` would break this behaviour.

Reply:
Er, um, having methods return different values for objects which hash the same isn't great, are there any other methods or properties which would return different values or is it just `get_is_in_weeks` (if so no biggie).

Reply:
I know it isn't great, presumably if we don't cache `get_is_in_weeks()` it should be okay though? `__str__()` is the only other one, unless you count e.g. `_copy()` and `__add__()` which return new `Duration`s, which may or may not be in weeks

Reply:
Sorry for being forgetful (too many PRs going on at the moment), could you remind me why we want `hash(P1W) == hash(P7D)`.

Reply:
No problem, it's because [the docs](https://docs.python.org/3/reference/datamodel.html#object.__hash__) say
> The only required property is that objects which compare equal have the same hash value

Reply:
Thanks for reminding me!

Ok, lets just keep this in the back of our minds for now.